src/smt/theory_str_noodler/theory_str_noodler.cpp

```diff
@@ -1972,6 +1976,10 @@ namespace smt::noodler {
         for (const auto& wi : this->m_word_diseq_todo_rel) {
 //            expr *const e = mk_eq_atom(wi.first, wi.second);
             expr_ref e(m.mk_not(ctx.mk_eq_atom(wi.first, wi.second)), m);
+            // e might not be internalized
```

Comment:
Do you know why e is not internalized? And can this also happen for equality?

Reply:
Do not know for sure. I have some guesses, but no evidence for that. It never happed for equality. Maybe because equations are handled differently that disequations and other constraints in z3.